src/exchange/group/delegation/group-delegation.controller.js

```diff
@@ -103,17 +124,46 @@ angular.module('Module.exchange.controllers').controller(
         });
     }
 
+    fetchAccountCreationOptions() {
+      this.loadingDomains = true;
+      return this.services.Exchange.fetchingAccountCreationOptions(
+        this.$routerParams.organization,
+        this.$routerParams.productId,
+      )
+        .then((accountCreationOptions) => {
+          const domains = [this.allDomainsOption];
+          _.forEach(accountCreationOptions.availableDomains, (domain) => {
```

Comment:
`forEach` + `push` can be transformed into `map`

Reply:
Yes, but I also need to find the selected item from the array. I need to do two things in one loop, to add all array contents into another and also to find the currently selected item in the array. Can I do both of these using the only map?